core/Pos/Core/Types.hs

```diff
@@ -248,11 +247,11 @@ type ProxySigHeavy a = ProxySignature EpochIndex a
 -- | Heavy delegation psk.
 type ProxySKHeavy = ProxySecretKey EpochIndex
 
--- | Some proxy secret key.
-type ProxySKEither = Either ProxySKLight ProxySKHeavy
-
--- | Map from issuers to heavy certs.
-type ProxySKHeavyMap = HashMap PublicKey ProxySKHeavy
+-- | Lightweight PSK or heavyweight PSK with real leader public key
```

Comment:
Okay, I grepped where it's used and now I see that it's used only for block creation.

> This is used to create a block header mainly.

So, not mainly, but only.

Probably it shouldn't be in `core` then, but somewhere in `Pos.Block` (perhaps `Pos.Block.Core.Main.Types`, near `BlockSignature`).

Secondly, all usages of it are `Maybe ProxySKEither`. So perhaps we don't need this type without `Maybe`. Also I never liked this `Maybe $ Either b c`, I think it should be a new datatype.